File: C++/Coins_and_Triangle.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int testCases;
    cin >> testCases;
    for (int j = 0; j < testCases; j++)
    {
        int height = 0;
        int number;
        cin >> number;
        int i = 1;
        while (number > 0)
        {
            if (number >= i)
            {
                number -= i;
                i++;
                height++;
            }
            else
            {
                break;
            }
        }
        cout << height << endl;
    }
}

File: PYTHON/Data Structure and Algorithms/Coins_and_Triangle.py
testCases = int(input())
for i in range(testCases):
    height = 0
    value = 1
    number = int(input())
    while(number > 0):
        if (number >= value):
            number -= value
            value += 1
            height += 1
        else:
            break
    print(height)
